transformer_engine/pytorch/attention.py

```diff
@@ -5090,19 +5089,15 @@ def forward(
                 output.reshape(batch_size * max_seqlen_q // cp_size, -1).transpose_2d()
                 output = output.reshape(batch_size, max_seqlen_q // cp_size, -1)
```

Comment:
I think the reshaping/transpose here is a bit fishy - I'll have to take another look at it (It's from my FA3 PR).